### DISCUSSION TITLE
post/golang-concurrent-programming-channel

### DISCUSSION THREAD

**Discussion by giscus** [nickxudotme/nickxudotme.github.io#4] [General]
# post/golang-concurrent-programming-channel

通道（Channel）  通道是什么，为什么使用通道  「不要通过共享内存来通信，而应该通过通信来共享内存」 通道可以在多个 goroutine 之间传递数据 一个通道相当于一个先进先出（FIFO）的队列。也就是说，通道中的各个元素值都是严格地按照发送的顺序排列的，先被发送通道的元素值一定会先被接收

https://nickxu.me/post/golang-concurrent-programming-channel

**Comment by aFlyBird0** [comment]
还可以加几条，比如一般由发送方关闭通道

**Reply by aFlyBird0** [reply]
for range 那个也能结合一下，在goroutine里去for range一个chan，然后当发送方关闭的时候，这个for自动就退出了
